services/mlmodel/tflitecpu/tflite_cpu.go

```diff
@@ -108,47 +109,38 @@ func NewTFLiteCPUModel(ctx context.Context, params *TFLiteConfig, name resource.
 
 // Infer takes the input map and uses the inference package to
 // return the result from the tflite cpu model as a map.
-func (m *Model) Infer(ctx context.Context, input map[string]interface{}) (map[string]interface{}, error) {
+func (m *Model) Infer(ctx context.Context, tensors ml.Tensors, input map[string]interface{}) (ml.Tensors, map[string]interface{}, error) {
 	_, span := trace.StartSpan(ctx, "service::mlmodel::tflite_cpu::Infer")
 	defer span.End()
-
-	outMap := make(map[string]interface{})
-	doInfer := func(input interface{}) (map[string]interface{}, error) {
-		outTensors, err := m.model.Infer(input)
-		if err != nil {
-			return nil, errors.Wrapf(err, "couldn't infer from model %q", m.Name())
-		}
-
-		// Fill in the output map with the names from metadata if u have them
-		// Otherwise, do output1, output2, etc.
-		n := int(math.Min(float64(len(m.metadata.Outputs)), float64(len(m.model.Info.OutputTensorTypes))))
-		for i := 0; i < n; i++ {
-			if m.metadata.Outputs[i].Name != "" {
-				outMap[m.metadata.Outputs[i].Name] = outTensors[i]
+	if input != nil {
+		m.logger.Warn("input maps for tflite_cpu.Infer are deprecated")
+	}
```

Comment:
@tahiyasalam tflite_cpu model in RDK won't accept map[string]interface{} inputs anymore - I think this is probably all right, as long as you don't have any non-RDK ml model servers that are still sending that data struct over.